cmd/grpcurl/grpcurl.go

```diff
@@ -426,6 +430,9 @@ func main() {
 		} else if *authority != "" {
 			opts = append(opts, grpc.WithAuthority(*authority))
 		}
+		if *user_agent != "" {
+			opts = append(opts, grpc.WithUserAgent(*user_agent))
```

Comment:
Oh, neat. While setting user agent, I think it would be good to add grpcurl version info in here, too.

 Maybe something like so:
```go
grpcurlUA := sanitizeUA("grpcurl/" + version)
if version == "dev build <no version set>" {
    grpcurlUA = "grpcurl/dev-build (no version set)"
}
if *userAgent != "" {
    grpcurlUA = sanitizeUA(*userAgent) + " " + grpcurlUA
}
append(opts, grpc.WithUserAgent(grpcurlUA))

// sanitizeUA should replace whitespace chars (space, tab, CR, LF, vertical tab)
// with underscores. Ideally, it would be a little smarter and recognize if the UA
// is already in valid format -- <product>/<version> (<extra>) -- and only replace
// characters that need to be replaced to preserve this.
```

Would you be up for making such a change?

Reply:
@jhump I've done a subset of the requested changes. The sanitization, however, I've skipped. Thinking a bit more about it, I feel that enforcing the format would be too restrictive; further, silently modifying the passed user-agent might be confusing. I can see use cases (e.g., testing) where that format is not followed. Thoughts?